src/components/settingsmenus/DateTimeMenu.js

```diff
@@ -0,0 +1,242 @@
+  // Return the string "+10:00" for the input 10.
+  // Should we laugh or cry?
+  const absOffset = Math.abs(offset);
+  const hours = Math.floor(absOffset);
+  const minutes = Math.floor((absOffset - hours) * 60);
+
+  return (
+    (offset >= 0 ? "+" : "-") +
+    (hours < 10 ? "0" + hours : hours) +
+    ":" +
+    (minutes < 10 ? "0" + minutes : minutes)
+  );
+};
+
+const getDateAndTimeStrings = function(dateTime, offset) {
+  // Takes a UTC dateTime string like "2019-03-21T10:58Z" and an
+  // offset in hours like 1. Returns a date and time in the timezone
+  // with that offset representing the same time.
+  if (!dateTime) {
+    return [null, null];
+  }
+
+  const utcTimestamp = new Date(dateTime).getTime();
+  const withOffset = utcTimestamp + 3600000 * offset;
+
+  const offsetDate = new Date(withOffset);
+
+  const iso = offsetDate.toISOString();
+  return [iso.slice(0, 10), iso.slice(11, 16)];
+};
+
+class DateTimeMenu extends Component {
+  constructor(props) {
+    super(props);
+    this.setTime = this.setTime.bind(this);
+  }
+
+  setDateTimeUsingLocalTimestamp(timestamp) {
+    timestamp -= this.props.currentTimezoneOffset * 3600000;
+    const newUtcDateTime = new Date(timestamp);
+    this.props.changeDateTime(newUtcDateTime.toISOString());
+  }
+
+  setTime(time) {
+    if (!time || !this.props.currentDate) return;
+
+    const fakeUtcDateTime = this.props.currentDate + "T" + time + "Z";
+    const timestamp = new Date(fakeUtcDateTime).getTime();
+    this.setDateTimeUsingLocalTimestamp(timestamp);
+  }
+
+  setDate(date) {
+    if (!date) return;
+    const fakeUtcDateTime =
+      date + "T" + (this.props.currentTime || "00:00") + "Z";
+    let timestamp = new Date(fakeUtcDateTime).getTime();
+    this.setDateTimeUsingLocalTimestamp(timestamp);
+  }
+
+  disabledRender() {
+    return (
+      <div style={{ padding: 20 }}>
+        <h4 style={{ padding: 0, margin: 0 }}>
+          Date/time settings &nbsp;
+          <button onClick={this.props.resetDateTime}>Reset</button>
+        </h4>
+        <hr />
+        <p>
+          The date and time have been set to a fixed value in this dashboard's
+          configuration, and can not be changed manually.
+        </p>
+        <button
+          className={styles.OKButton}
+          onClick={() => {
+            this.props.closeSettingsMenu();
+          }}
+        >
+          OK
+        </button>
+      </div>
+    );
+  }
+
+  render() {
+    if (this.props.disableDateTimeSettings) {
+      return this.disabledRender();
+    }
+
+    return (
+      <div style={{ padding: 20 }}>
+        <h4 style={{ padding: 0, margin: 0 }}>
+          Date/time settings &nbsp;
+          <button onClick={this.props.resetDateTime}>Reset</button>
+        </h4>
+        <hr />
+        <div className={styles.DateTimePicker}>
+          <div>
+            <h5>Date (e.g. "23/12/2018")</h5>
+            <input
+              type="date"
+              name="date"
```

Comment:
de value=currentDate lijkt niet te werken nadat de user op "reset" klikt. In de redux store lijkt dit wel te werken.